### PR TITLE
paging-calculator: init at 0.1.2

### PR DESCRIPTION
**Pull request by phip1611**
###### Description of changes

This adds <https://github.com/phip1611/paging-calculator> to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/paging-calculator/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  version = "0.1.2";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-9DbpfJbarWXYGNzTqfHHSaKFqSJ59E/VhOhuMUWk8ho=";
+  };
+
+  cargoSha256 = "sha256-IfOhJwR5eRHeeAbEZ8zeUVojQXtrYHdzAeht/lvdlUQ=";
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/development/tools/paging-calculator/default.nix:17]:
```suggestion
  meta = {
```

File: pkgs/development/tools/paging-calculator/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      paging-calculator is a CLI utility written in Rust that helps you find the indices that a virtual
+      address will have on different architectures or paging implementations.
+
+      It takes a (virtual) address in hexadecimal format and shows you which index will be used for
+      what page-table level. It can be installed with $ cargo install paging-calculator.
+    '';
+    homepage = "https://github.com/phip1611/paging-calculator";
+    changelog = "https://github.com/phip1611/paging-calculator/blob/v${version}/CHANGELOG.md";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ phip1611 ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/paging-calculator/default.nix:29]:
```suggestion
    license = with lib.licenses; [ mit ];
    maintainers = with lib.maintainers; [ phip1611 ];
```

File: pkgs/development/tools/paging-calculator/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  };
+}
+
```
Comment (AndersonTorres) [inline pkgs/development/tools/paging-calculator/default.nix:32]:
```suggestion
```

File: pkgs/development/tools/paging-calculator/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "paging-calculator";
+  version = "0.1.2";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-9DbpfJbarWXYGNzTqfHHSaKFqSJ59E/VhOhuMUWk8ho=";
```
Comment (AndersonTorres) [inline pkgs/development/tools/paging-calculator/default.nix:12]:
```suggestion
    hash = "sha256-9DbpfJbarWXYGNzTqfHHSaKFqSJ59E/VhOhuMUWk8ho=";
```

File: pkgs/development/tools/paging-calculator/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  cargoSha256 = "sha256-IfOhJwR5eRHeeAbEZ8zeUVojQXtrYHdzAeht/lvdlUQ=";
```
Comment (AndersonTorres) [inline pkgs/development/tools/paging-calculator/default.nix:15]:
```suggestion
  cargoHash = "sha256-IfOhJwR5eRHeeAbEZ8zeUVojQXtrYHdzAeht/lvdlUQ=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by phip1611** [comment]
please recheck @AndersonTorres

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**